### PR TITLE
[Buttons] New Dynamic Type curves

### PR DESCRIPTION
**Pull request by ianegordon**
Add support for the new dynamic type curves.

Note that the new curve values are identical to the old curve values, just using the new API.

Large / Default:
MDCButton-L

XXS:
MDCButton-XXS

XXXL:
MDCButton-XXXL

### CODE REVIEW

File: components/Buttons/src/MDCButton.m [hunk 1]
```diff
@@ -819,7 +819,11 @@ - (void)updateTitleFont {
 
   if (_mdc_adjustsFontForContentSizeCategory) {
     // Dynamic type is enabled so apply scaling
-    font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton scaledForDynamicType:YES];
+    if (font.mdc_scalingCurve && !_mdc_legacyFontScaling) {
+      font = [font mdc_scaledFontForCurrentSizeCategory];
```
Comment (yarneo) [inline components/Buttons/src/MDCButton.m:823]: is there no need to provide what style to give the font?

**Review by yarneo — APPROVED** [review]

File: components/Buttons/src/MDCButton.m (hunk 1 above)
Comment (ianegordon) [reply components/Buttons/src/MDCButton.m:823]: We do a little earlier in the Sample.
This font is set by the themer, which takes it from containerScheme.typographyScheme.button.

In the case of the new scheme, that font will have an mdc_scalingCurve attached.

Comment (yarneo) [reply components/Buttons/src/MDCButton.m:823]: ah awesome

File: components/Buttons/examples/ButtonsDynamicTypeViewController.swift [hunk 2]
```diff
@@ -47,29 +47,43 @@ class ButtonsDynamicTypeViewController: UIViewController {
...
-    flatButtonDynamic.setTitleColor(titleColor, for: .normal)
-    flatButtonDynamic.setBackgroundColor(backgroundColor, for: .normal)
-    flatButtonDynamic.setTitle("Dynamic", for: UIControlState())
-    flatButtonDynamic.sizeToFit()
-    flatButtonDynamic.translatesAutoresizingMaskIntoConstraints = false
-    flatButtonDynamic.addTarget(self, action: #selector(tap), for: .touchUpInside)
-    flatButtonDynamic.mdc_adjustsFontForContentSizeCategory = true
-    view.addSubview(flatButtonDynamic)
+    containerScheme.typographyScheme = MDCTypographyScheme.init(defaults: .material201902)
+    let flatButtonDynamicM2 = MDCButton()
```
Comment (jverkoey) [inline components/Buttons/examples/ButtonsDynamicTypeViewController.swift]: Please consider an alternative suffix :)

File: components/Buttons/src/MDCButton.h [hunk 3]
```diff
@@ -118,6 +118,13 @@
 @property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
     BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;
 
+/**
+ Enable legacy font scaling curves for Dynamic Type
```
Comment (jverkoey) [inline components/Buttons/src/MDCButton.h]:
```suggestion
 Enable legacy font scaling curves for Dynamic Type.
```

File: components/Buttons/src/MDCButton.h [hunk 4]
```diff
@@ -118,6 +118,13 @@
...
+
```
Comment (jverkoey) [inline components/Buttons/src/MDCButton.h:126]: Please explain what "legacy font scaling curves" means here.

**Review by jverkoey — APPROVED** [review]

File: components/Buttons/examples/ButtonsDynamicTypeViewController.swift (hunk 2 above)
Comment (ianegordon) [reply components/Buttons/examples/ButtonsDynamicTypeViewController.swift]: Done.

**Comment by ianegordon** [comment]
CLA warning is not accurate as @jverkoey is a Googler, even if his @google.com email address wasn't attached to his recommendation / commit.

File: components/Buttons/src/MDCButton.h (hunk 4 above)
Comment (ianegordon) [reply components/Buttons/src/MDCButton.h:126]: Done.

**Comment by romoore** [comment]
CLA warning is accurate since the commits came from multiple sources. To ensure the original author (and not only the submitter) wished to merge the PR, the bot sets `cla: No` as a reminder for PR authors to get consent from all authors and document the same.

**Pull request merged**